pkg/piecestore/rpc/client/readerwriter.go

```diff
@@ -44,15 +70,35 @@ type StreamReader struct {
 }
 
 // NewStreamReader creates a StreamReader for reading data from the piece store server
-func NewStreamReader(stream pb.PieceStoreRoutes_RetrieveClient) *StreamReader {
+func (signer *Client) NewStreamReader(stream pb.PieceStoreRoutes_RetrieveClient) *StreamReader {
 	return &StreamReader{
 		stream: stream,
 		src: utils.NewReaderSource(func() ([]byte, error) {
-			msg, err := stream.Recv()
+
+			// TODO: What does the message look like?
+			sig, err := signer.sign(context.Background(), []byte{'m', 's', 'g'})
```

Comment:
Same comment about the context here too. Perhaps, the NewStreamReader should take a context and pass it to `sign()`.